packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -10,39 +10,135 @@ import 'framework.dart';
 import 'ticker_provider.dart';
 import 'transitions.dart';
 
+// Internal representation of a child that, now or in the past, was set on the
+// AnimatedChildSwitcher.child field, but is now in the process of
+// transitioning. The internal representation includes fields that we don't want
+// to expose to the public API (like the controller).
 class _AnimatedChildSwitcherChildEntry {
-  _AnimatedChildSwitcherChildEntry(this.widget, this.controller, this.animation);
+  _AnimatedChildSwitcherChildEntry({
+    @required this.widgetChild,
+    @required this.transition,
+    @required this.animation,
+    @required this.controller,
+  })  : assert(transition != null),
+        assert(animation != null),
+        assert(controller != null);
 
-  Widget widget;
+  final Animation<double> animation;
+
+  // The currently built transition for this child.
+  Widget transition;
 
+  // The animation controller for the child's transition.
   final AnimationController controller;
-  final Animation<double> animation;
+
+  // The widget's child at the time this entry was created or updated.
+  Widget widgetChild;
```

Comment:
please keep the arguments, asserts, and fields in the same order.

Reply:
Sorry, got a little messed up there... Done.